File: Problems/202_Happy_Number.cpp
class Solution {
public:
    bool isHappy(int n){
        unordered_set<int> seen;
        while(n!=1){
            int newN = 0;
            while(n){
                newN += pow(n%10, 2);
                n /=10;
            }
            n = newN;
            if(seen.count(n)) return false;
            seen.insert(n);
        }
        return true;
    }
};

File: Problems/202_Happy_Number.py
class Solution:
    def isHappy(self, n: int) -> bool:
        seen = {n}
        
        while True:
            total = 0
            while n:
                total += (n % 10)**2
                n //= 10
            if total == 1:
                return True
            if total in seen:
                return False
            seen.add(total)
            n = total